src/jit/lsra.cpp

```diff
@@ -6836,7 +6919,20 @@ void LinearScan::freeRegister(RegRecord* physRegRecord)
             // we wouldn't unnecessarily link separate live ranges to the same register.
             if (nextRefPosition == nullptr || RefTypeIsDef(nextRefPosition->refType))
             {
+#ifdef _TARGET_ARM_
+                if (assignedInterval->registerType == TYP_DOUBLE && !genIsValidDoubleReg(physRegRecord->regNum))
+                {
+                    // Don't call unassignPhysReg() again, because two FP registers of a double reister are already
+                    // unassigned together when this function is called with valid double register number.
+                    assert(assignedInterval->assignedReg != physRegRecord);
```

Comment:
>assert(assignedInterval->assignedReg != physRegRecord); [](start = 20, length = 55)

I believe that this assert should be stronger. In fact, I think that it is problematic if this method is ever called with the other half of a `TYP_DOUBLE` register when the interval being ejected is itself `TYP_DOUBLE`. That is, such registers should always be freed together (and only the half that is the valid double register should be added to the set of registers to free).